smolyay/basis.py

```diff
@@ -128,6 +137,35 @@ def __call__(self,x):
                 answer += math.comb(self._n,2*k)*((x**2 - 1)**k)*(x**(self._n-2*k))
             return answer
 
+    def derivative(self, x):
```

Comment:
You should add an abstract `derivative` method to the `BasisFunction` that defines this interface. Then, every basis function is responsible for implementing the method.